packages/engine-core/src/library/LibraryEngine.ts

```diff
@@ -100,6 +101,29 @@ export class LibraryEngine implements Engine {
     initHooks(this)
   }
 
+  async transaction(action: 'start', options?: Tx.Options): Promise<Tx.Info>
+  async transaction(action: 'commit', info: Tx.Info): Promise<undefined>
+  async transaction(action: 'rollback', info: Tx.Info): Promise<undefined>
+  async transaction(action: any, arg?: any) {
+    await this.start()
+
+    if (action === 'start') {
+      const jsonOptions = JSON.stringify({
+        max_wait: arg?.maxWait ?? 2000, // default
+        timeout: arg?.timeout ?? 5000, // default
+      })
+
+      const result = await this.engine?.startTransaction(jsonOptions, '{}')
+      return this.parseEngineResponse<Tx.Info>(result)
+    } else if (action === 'commit') {
+      await this.engine?.commitTransaction(arg.id, '{}')
+    } else if (action === 'rollback') {
+      await this.engine?.rollbackTransaction(arg.id, '{}')
+    }
```

Comment:
How is it possible that `engine?` might not be defined?

Reply:
I think that it boils down from the fact that we do async stuff to init the engine. Maybe we should add some type assertion logic into `start` to prevent that.